### PR TITLE
subcache filled and emptied but not used

### PR DESCRIPTION
**Pull request by kzangeli**
cache created, emptied and recreated.
Not used, still a lot missing ...

### CODE REVIEW

File: src/lib/mongoBackend/mongoSubCache.cpp [hunk 1]
```diff
@@ -13,10 +40,237 @@ using std::auto_ptr;
...
+
+
+
+/* ****************************************************************************
+*
+* EntityInfo::present - 
+*/
+void EntityInfo::present(const std::string& prefix)
+{
+  LM_F(("%sid:    regex describing EntityId::id (the original string is not saved)", prefix.c_str()));
```
Comment (fgalan) [inline src/lib/mongoBackend/mongoSubCache.cpp]: Maybe LM_M would me this more easy to "catch" to be removed at the end? (I think that LM_F are used for present() methods, which is ok, but probably I don't remembering well...).

Comment (fgalan) [inline src/lib/mongoBackend/mongoSubCache.cpp]: Argghh... ignore my comment... I have just realize that the LM_F are inside within a present() method :$

NTC

File: src/lib/mongoBackend/mongoSubCache.cpp [hunk 2]
```diff
@@ -26,28 +280,155 @@ static void subCacheItemUpdate(const char* tenant, BSONObj* subP)
...
-  int64_t                   expiration        = subP->hasField(CSUB_EXPIRATION)? subP->getField(CSUB_EXPIRATION).Long() : 0;
-  int64_t                   throttling        = subP->hasField(CSUB_THROTTLING)? subP->getField(CSUB_THROTTLING).Long() : -1;
+  //
+  // 03. Create CachedSubscription
+  //
+  CachedSubscription* cSubP = (CachedSubscription*) calloc(1, sizeof(CachedSubscription));
+
+  if (cSubP == NULL)
+  {
+    LM_E(("Runtime Error (cannot allocate memory for a cached subscription: %s)", strerror(errno)));
```
Comment (fgalan) [inline src/lib/mongoBackend/mongoSubCache.cpp]: Probably this should be Fatal Error and invoke a orionExit(). The broker shouldn't continue running if memory has been exahusted.

However, don't need to fix this right now. Just use a "FIXME" above this line to indicate that behaviour for a next round.

**Comment by fgalan** [comment]
Mostly LGTM (only a minor comment). However, before merging, just to check... is the code in this branch passin the "jmeter test" (DEBUG & RELEASE, several times each one) without breaking?

(For next PRs, probably this should be mentioned in the PR body :)

File: src/lib/mongoBackend/mongoSubCache.cpp (hunk 2 above)
Comment (kzangeli) [inline src/lib/mongoBackend/mongoSubCache.cpp]: Sure, no real meaning to continue when we have no more RAM ...
Fixing it in my current branch, as to save time.

**Comment by kzangeli** [comment]
I will *not* send a PR without the branch running OK on my local machine, of course not :-)

File: src/lib/mongoBackend/mongoSubCache.cpp (hunk 2 above)
Comment (fgalan) [inline src/lib/mongoBackend/mongoSubCache.cpp]: Ok. NTC

**Comment by fgalan** [comment]
LGTM

**Pull request merged**